File: Models/Movie.cs
using System.Text.Json.Serialization;

namespace FilmVault.Models
{
    public class Movie
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("title")]
        public string Title { get; set; }

        [JsonPropertyName("overview")]
        public string Overview { get; set; }

        [JsonPropertyName("poster_path")]
        public string PosterPath { get; set; }

        [JsonPropertyName("vote_average")]
        public double VoteAverage { get; set; }

        [JsonPropertyName("release_date")]
        public string ReleaseDate { get; set; }

        // Derived property, not part of TMDB JSON
        public string FullPosterPath =>
            string.IsNullOrEmpty(PosterPath)
                ? ""
                : $"https://image.tmdb.org/t/p/w500{PosterPath}";
    }
    public class MovieSearchResult
    {
        [JsonPropertyName("results")]
        public List<Movie> Results { get; set; }
    }
}

File: client/src/types/Movie.ts
/**
 * Movie interface for representing movie data.
 * This interface is used throughout the application to ensure type safety
 */
export interface Movie {
  id: number;
  title: string;
  overview: string;
  posterPath: string | null;
  voteAverage: number;
  releaseDate: string | null;
  fullPosterPath: string | null;
}

/**
 * RawMovie interface for representing raw movie data from the TMDb API.
 */
export interface RawMovie {
  id: number;
  title: string;
  overview: string;
  poster_path: string | null;
  vote_average: number;
  release_date: string | null;
}
